contracts/game/src/game/renderer.cairo

```diff
@@ -48,37 +63,295 @@ fn create_svg(internals: ByteArray) -> ByteArray {
+        );
+
+    let mut _item_prefix_2 = Default::default();
+    _item_prefix_2
+        .append_word(
+            ImplItemNaming::prefix2_to_string(specials.special2),
+            U256BytesUsedTraitImpl::bytes_used(
+                ImplItemNaming::prefix1_to_string(specials.special2).into()
+            )
+                .into()
+        );
+
+    let mut _item_suffix = Default::default();
+    _item_suffix
+        .append_word(
+            ImplItemNaming::suffix_to_string(specials.special3),
+            U256BytesUsedTraitImpl::bytes_used(
+                ImplItemNaming::prefix1_to_string(specials.special3).into()
+            )
+                .into()
+        );
+
+    if (specials.special1 != 0) {
+        _item_prefix_1 = _item_prefix_1 + " ";
+    } else {
+        _item_prefix_1 = "";
+    }
+
+    if (specials.special2 != 0) {
+        _item_prefix_2 = _item_prefix_2 + " ";
+    } else {
+        _item_prefix_2 = "";
+    }
+
+    if (specials.special3 != 0) {
+        _item_suffix = " " + _item_suffix;
+    } else {
+        _item_suffix = "";
+    }
+
+    _item_prefix_1 + _item_prefix_2 + _item_name + _item_suffix
+}
+
+fn create_metadata(
+    adventurer_id: felt252, adventurer: Adventurer, adventurerMetadata: AdventurerMetadata, bag: Bag
+) -> ByteArray {
     let rect = create_rect();
 
     let logo_element = "<g transform='translate(25,25) scale(4)'>" + logo() + "</g>";
 
-    // Update text elements
-    let name = create_text("John Doe", "117", "117.136", "32", "middle");
-    let id = create_text("#1234", "123", "61.2273", "24", "middle");
-    let level = create_text("LVL 42", "208.008", "61.2273", "24", "middle");
-    let health = create_text("100/100 HP", "453.527", "58.2727", "20", "right");
-    let gold = create_text("1000 GLD", "475.09", "93.2727", "20", "right");
-
-    // Stats
-    let str = create_text("8 STR", "511.672", "128.273", "20", "right");
-    let dex = create_text("8 DEX", "510.891", "163.273", "20", "right");
-    let int = create_text("7 INT", "517.766", "198.273", "20", "right");
-    let vit = create_text("5 VIT", "518.566", "233.273", "20", "right");
-    let wis = create_text("9 WIS", "512.863", "268.273", "20", "right");
-    let cha = create_text("10 CHA", "497.707", "303.273", "20", "right");
-    let luck = create_text("2 LUCK", "496.594", "338.273", "20", "right");
-
-    // Equipment sections
-    let equipped_header = create_text("Equipped", "30", "183.136", "32", "middle");
-    let bag_header = create_text("Bag", "30", "600.136", "32", "middle");
+    let mut _name = Default::default();
+    _name
+        .append_word(
+            adventurerMetadata.name,
+            U256BytesUsedTraitImpl::bytes_used(adventurerMetadata.name.into()).into()
+        );
+
+    let _adventurer_id = format!("# {}", adventurer_id);
+    let _level = format!("LVL {}", adventurer.get_level());
+
+    let stat_boosts = adventurer.equipment.get_stat_boosts(adventurerMetadata.start_entropy);
```

Comment:
This call will return the stat boosts being provided by the adventurers equipment. For the stats, you want to use adventurer.stats.strength, adventurer.stats.vitality, etc

Reply:
fixed